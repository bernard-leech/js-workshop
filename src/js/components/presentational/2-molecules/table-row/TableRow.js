// @flow
import React from 'react'

import StatusPill from '../../1-atoms/StatusPill'

import type { TableRowType } from './TableRow.types'

function TableRow (props: TableRowType) {
  const {
    heading,
    money,
    date,
    status
  } = props
  return (
    <div>
      <div>{heading}</div>
      <div>{money}</div>
      <div>{date}</div>
      <StatusPill status={status} children={status} />
    </div>
  )
}

export default TableRow
