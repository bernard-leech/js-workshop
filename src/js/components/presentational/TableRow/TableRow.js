// @flow
import React from 'react'
import classNames from 'classnames'

import type { TableRowType } from './TableRow.types'

function TableRow (props: TableRowType) {
  const {
    heading,
    date,
    money,
    status,
    links
  } = props

  return (
    <div>
      {heading}
      {date}
      {money}
      {status}
    </div>
  )
}

export default TableRow
