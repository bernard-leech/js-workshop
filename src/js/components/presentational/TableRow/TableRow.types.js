// @flow

export type TableRowType = {
  heading?: string
}
