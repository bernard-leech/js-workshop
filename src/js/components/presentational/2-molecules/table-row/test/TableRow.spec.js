import React from 'react'
import TestRenderer from 'react-test-renderer'

import { props } from '../test/TableRow.mock'

import TableRow from '../TableRow'

it('renders correctly', () => {
  const tree = TestRenderer
    .create(<TableRow />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
