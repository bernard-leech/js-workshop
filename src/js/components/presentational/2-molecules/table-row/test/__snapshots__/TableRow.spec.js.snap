// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div />
  <div />
  <div />
  <div
    className="status-pill"
  />
</div>
`;
